core-web/libs/portlets/dot-ema/src/lib/components/edit-ema-persona-selector/edit-ema-persona-selector.component.ts

```diff
@@ -0,0 +1,91 @@
+import { CommonModule } from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    EventEmitter,
+    Input,
+    OnInit,
+    Output,
+    inject
+} from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+import { AutoCompleteModule } from 'primeng/autocomplete';
+import { AvatarModule } from 'primeng/avatar';
+import { BadgeModule } from 'primeng/badge';
+import { ButtonModule } from 'primeng/button';
+import { ConfirmDialogModule } from 'primeng/confirmdialog';
+import { ListboxModule } from 'primeng/listbox';
+import { OverlayPanelModule } from 'primeng/overlaypanel';
+
+import { DotPersona } from '@dotcms/dotcms-models';
+import { DotAvatarDirective, DotMessagePipe } from '@dotcms/ui';
+
+import { DotPageApiService } from '../../services/dot-page-api.service';
+
+@Component({
+    selector: 'dot-edit-ema-persona-selector',
+    standalone: true,
+    imports: [
+        CommonModule,
+        ButtonModule,
+        AvatarModule,
+        OverlayPanelModule,
+        DotAvatarDirective,
+        DotMessagePipe,
+        ListboxModule,
+        BadgeModule,
+        ConfirmDialogModule,
+        FormsModule,
+        AutoCompleteModule,
+        FormsModule
+    ],
+
+    templateUrl: './edit-ema-persona-selector.component.html',
+    styleUrls: ['./edit-ema-persona-selector.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class EditEmaPersonaSelectorComponent implements OnInit {
+    private readonly pageApiService = inject(DotPageApiService);
+    private personas: DotPersona[];
+
+    @Input() pageID: string;
+    @Input() value: DotPersona;
+
+    @Output() selected: EventEmitter<DotPersona & { pageID: string }> = new EventEmitter();
+
+    filteredPersonas: DotPersona[] = [];
+
+    ngOnInit(): void {
+        this.pageApiService
+            .getPersonas({
+                pageID: this.pageID,
+                // TODO: when we update to PrimeNG 17 we can do this async
```

Comment:
Please create an issue ticket for TODOs. No TODO must be left in production code.